src/components/Views/ObservationsView/ObservationsView.js

```diff
@@ -73,80 +136,141 @@ I18n.translations = {
-        </TouchableOpacity>
-      </View>
-      <View style={styles.container}>
-        <Text style={styles.header}>{I18n.t('observations.title')}</Text>
-        <SectionList
-          style={{ flex: 1, flexDirection: 'column' }}
+    observations.forEach(o => {
+      const createdMoment = moment(o.created);
+      const thisWeek = moment().startOf('week');
+      const thisMonth = moment().startOf('month');
+      const thisYear = moment().startOf('year');
+      if (createdMoment.isSameOrAfter(thisWeek)) {
+        label = `${I18n.t('observations.this_week')}`;
+      } else if (createdMoment.isSameOrAfter(thisMonth)) {
+        label = `${I18n.t('observations.this_month')}`;
+      } else if (createdMoment.isSameOrAfter(thisYear)) {
+        label = createdMoment.format('MMMM');
+      } else {
+        label = createdMoment.format('MMMM YYYY');
+      }
+    });
+
+    const sortedObservations = orderBy(observations, ['created'], ['desc']);
+
+    const keyExtractor = item => item.id.toString();
+    const handleItemPress = item => {
+      this.props.selectObservation(item);
+      this.props.goToObservationDetail();
+    };
+
+    return (
+      <View
+        style={{
+          flex: 1
+        }}
+      >
+        <FlatList
+          scrollEnabled
+          ListHeaderComponent={
+            <View
+              style={{
+                flexDirection: 'column'
+              }}
+            >
+              <View style={styles.header}>
+                <TouchableOpacity
+                  onPress={this.props.closeRightDrawer}
+                  style={styles.closeDrawerButton}
+                >
+                  <LeftChevron color="#a5a5a4" name="chevron-left" size={30} />
+                </TouchableOpacity>
+                <View
+                  style={{
+                    flexDirection: 'column'
+                  }}
+                >
+                  <TouchableOpacity style={styles.syncButtonOuterCircle}>
+                    <View style={styles.syncButtonInnerCircle}>
+                      <SyncIcon
+                        color={WHITE}
+                        name="bolt"
+                        size={20}
+                        style={{ transform: [{ rotate: '15deg' }] }}
+                      />
+                    </View>
+                  </TouchableOpacity>
+                </View>
+                <TouchableOpacity>
+                  <SettingsIcon color="#a5a5a4" name="settings" size={30} />
+                </TouchableOpacity>
+              </View>
+              <View style={styles.headerBottom}>
+                <Text
+                  style={{ fontWeight: '700', fontSize: 12, color: 'black' }}
+                >
+                  {I18n.t('observations.all_observations')}
+                </Text>
+                <Text style={{ fontSize: 12 }}>
+                  {I18n.t('observations.view_by')}
+                </Text>
+              </View>
+              {this.state.showSyncTip ? (
```

Comment:
maybe just `!!this.state.showSyncTip && [component]`?

Reply:
oooh good call! shall change that